### PR TITLE
Add docstrings

### LINKED ISSUE #17: Add docstring documentation
Code needs documentation as generated by docstrings like the other libraries.  When this gets done, the link in the README should also be updated!

### PR DESCRIPTION
**Pull request by tekktrik**
Resolves #17 by adding docstring documentation, also updates supporting files so that Sphinx will build and show them as documentation.

### CODE REVIEW

**Comment by tekktrik** [comment]
Looks like the `asyncio.stream.py` has a bunch of `yield`s and `return`s mixed together, so I can hold off until it's possible for a single commit fix or workaround for them as opposed to doing it one by one

**Comment by tekktrik** [comment]
Hey @dhalbert, any update on this?  I know you're working on asyncio stuff so just wanted to check how this fits in, or if I should expect to update it as you make changes.

**Comment by dhalbert** [comment]
I think we can proceed on this. It looks like you found the places to change from `yield`. There are going to be some more changes re #22 but I don't think I will do that immediately.

File: asyncio/core.py [hunk 1]
```diff
@@ -232,6 +257,11 @@ def run_until_complete(main_task=None):
 
 # Create a new task from a coroutine and run it until it finishes
 def run(coro):
+    """Create a new task from the given coroutine and run it until it completes.
+
+    Returns the value returned be *coro*.
```
Comment (FoamyGuy) [inline asyncio/core.py:262]: should this be `...returned by *coro*` ?

File: asyncio/event.py [hunk 2]
```diff
@@ -33,15 +46,23 @@ def set(self):
...
+        This is a coroutine.
+        """
+
         if not self.state:
             # Event not set, put the calling task on the event's waiting queue
             self.waiting.push_head(core.cur_task)
             # Set calling task's data to the event's queue so it can be removed if needed
             core.cur_task.data = self.waiting
-            yield
+            await core.sleep(0)
```
Comment (FoamyGuy) [inline asyncio/event.py:65]: I am still wrapping my head around this library and the await / yield keywords and things. This change may be normal and expected, but just wanted to note it is the only thing I've come across so far that isn't docstrings.

File: asyncio/funcs.py [hunk 3]
```diff
@@ -10,16 +10,33 @@
...
 
 async def wait_for(aw, timeout, sleep=core.sleep):
+    """Wait for the *aw* awaitable to complete, but cancel if it takes longer
+    than *timeout* seconds. If *aw* is not a task then a task will be created
+    from it.
+
+    If a timeout occurs, it cancels the task and raises ``asyncio.TimeoutError``:
+    this should be trapped by the caller.
+
+    Returns the return vaalue of *aw*.
```
Comment (FoamyGuy) [inline asyncio/funcs.py:30]: minor typo, extra `a` in `vaalue`

File: asyncio/stream.py [hunk 4]
```diff
@@ -141,6 +219,15 @@ async def _serve(self, s, cb):
 # Helper function to start a TCP stream server, running as a new task
 # TODO could use an accept-callback on socket read activity instead of creating a task
 async def start_server(cb, host, port, backlog=5):
+    """Start a TCP server on the given *host* and *port*. The *cb* callback wil be
```
Comment (FoamyGuy) [inline asyncio/stream.py:222]: small typo, missing an `l` in `will`

**Review by FoamyGuy — CHANGES_REQUESTED** [review]
Thanks for imporoving the docs here @tekktrik. I found a few small typos to fix but the majority of this looks good to me. 

I noticed a few places where there are real code changes involving `await core.sleep(0)`. I don't have enough experience yet to review those one way or another. But thought it was worth mentioning since they seem to be the only changes outside of docstrings in this PR.

File: asyncio/core.py (hunk 1 above)
Comment (tekktrik) [reply asyncio/core.py:262]: Ah, yes, thank you@!

File: asyncio/stream.py (hunk 4 above)
Comment (tekktrik) [reply asyncio/stream.py]: Thank you!

**Comment by dhalbert** [comment]
Could you test at least some of the examples in the async guide: https://learn.adafruit.com/cooperative-multitasking-in-circuitpython with these changes? Thanks! I will take a final look after that.

**Comment by FoamyGuy** [comment]
I will test the learn guide examples with this branch in the afternoon after the weekly meeting.

**Comment by FoamyGuy** [comment]
I tested many of the examples from the learn guide using a Feather RP2040 with some LEDs and buttons wired up. For the neopixel rainbow animation one I used a NeoTrellis and slightly modified the code to read from one row of the matrix. 

Everything appears to be functioning as expected.

Ready for review when you have time @dhalbert

**Comment by tekktrik** [comment]
@dhalbert is this still good to go?  Looks like it passed  the hardware sanity check.

**Review by dhalbert — APPROVED** [review]
I think this is good. Sorry for the delay. Thanks to @FoamyGuy for testing.

**Pull request merged**